source/common/upstream/upstream_impl.h

```diff
@@ -1052,6 +1052,15 @@ class ClusterInfoImpl : public ClusterInfo,
   Http::Http2::CodecStats& http2CodecStats() const override;
   Http::Http3::CodecStats& http3CodecStats() const override;
   Http::ClientHeaderValidatorPtr makeHeaderValidator(Http::Protocol protocol) const override;
+  const ClusterProto::HappyEyeballsConfig happyEyeballsConfig() const override {
+    if (happy_eyeballs_config_ == nullptr) {
+      ClusterProto::HappyEyeballsConfig default_config;
```

Comment:
this constructs a new default_config every time happyEyeballsConfig is called.  Can we just initialize happy_eyeballs_config_ at construction time?

Reply:
changed to an optional object and changes the behavior that it doesn't create a default config when not provided. In that case, it will take the old path.